circuits/cpp/barretenberg/cpp/src/barretenberg/honk/proof_system/eccvm_verifier.cpp

```diff
@@ -0,0 +1,264 @@
+    commitments.lookup_read_counts_1 =
+        transcript.template receive_from_prover<Commitment>(commitment_labels.lookup_read_counts_1);
+
+    // Get challenge for sorted list batching and wire four memory records
+    auto [eta, gamma] = transcript.get_challenges("beta", "gamma");
+    relation_parameters.gamma = gamma;
+    auto eta_sqr = eta * eta;
+    relation_parameters.eta = eta;
+    relation_parameters.eta_sqr = eta_sqr;
+    relation_parameters.eta_cube = eta_sqr * eta;
+    relation_parameters.eccvm_set_permutation_delta =
+        gamma * (gamma + eta_sqr) * (gamma + eta_sqr + eta_sqr) * (gamma + eta_sqr + eta_sqr + eta_sqr);
+    relation_parameters.eccvm_set_permutation_delta = relation_parameters.eccvm_set_permutation_delta.invert();
+
+    // Get commitment to permutation and lookup grand products
+    commitments.lookup_inverses =
+        transcript.template receive_from_prover<Commitment>(commitment_labels.lookup_inverses);
+    commitments.z_perm = transcript.template receive_from_prover<Commitment>(commitment_labels.z_perm);
+
+    // Execute Sumcheck Verifier
+    auto sumcheck = SumcheckVerifier<Flavor>(circuit_size);
+
+    std::optional sumcheck_output = sumcheck.verify(relation_parameters, transcript);
+
+    // If Sumcheck does not return an output, sumcheck verification has failed
+    if (!sumcheck_output.has_value()) {
+        return false;
+    }
+
+    auto [multivariate_challenge, purported_evaluations] = *sumcheck_output;
+
+    // Execute Gemini/Shplonk verification:
+
+    // Construct inputs for Gemini verifier:
+    // - Multivariate opening point u = (u_0, ..., u_{d-1})
+    // - batched unshifted and to-be-shifted polynomial commitments
+    auto batched_commitment_unshifted = GroupElement::zero();
+    auto batched_commitment_to_be_shifted = GroupElement::zero();
+    const size_t NUM_POLYNOMIALS = Flavor::NUM_ALL_ENTITIES;
+    // Compute powers of batching challenge rho
+    FF rho = transcript.get_challenge("rho");
+    std::vector<FF> rhos = pcs::gemini::powers_of_rho(rho, NUM_POLYNOMIALS);
+
+    // Compute batched multivariate evaluation
+    FF batched_evaluation = FF::zero();
+    size_t evaluation_idx = 0;
+    for (auto& value : purported_evaluations.get_unshifted()) {
+        batched_evaluation += value * rhos[evaluation_idx];
+        ++evaluation_idx;
+    }
+    for (auto& value : purported_evaluations.get_shifted()) {
+        batched_evaluation += value * rhos[evaluation_idx];
+        ++evaluation_idx;
+    }
+
+    // Construct batched commitment for NON-shifted polynomials
+    size_t commitment_idx = 0;
+    for (auto& commitment : commitments.get_unshifted()) {
+        // very lazy point at infinity check. not complete. fix.
+        if (commitment.y != 0) {
+            batched_commitment_unshifted += commitment * rhos[commitment_idx];
+        } else {
+            std::cout << "point at infinity (unshifted)" << std::endl;
```

Comment:
We sometimes get issues from using stdio with wasm, so it's better to use info, which is changed according to the build

Reply:
done